From 6a55811a6d6baac4f270901078611adb589b1aaf Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 10 May 2016 16:45:13 +0200
Subject: [PATCH] Fix random build failure

uic and moc doesn't create output directory automatically and it could
be missing when uic is executed before gcc creates any object files as
src/*.o

OE @ ~/build/wpb/webos-ports/tmp-glibc/work/i586-webos-linux/phonesim/1.20-r0/build $ rm -rf src/

OE @ ~/build/wpb/webos-ports/tmp-glibc/work/i586-webos-linux/phonesim/1.20-r0/build $ /OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/uic ../phonesim-1.20/src/controlbase.ui -o src/ui_controlbase.h
Could not create output file

Fixes phonesim randomly failing with:
OE @ ~/build/wpb/webos-ports/tmp-glibc/work/i586-webos-linux/phonesim/1.20-r0/build $ cat ../temp/log.do_compile
DEBUG: SITE files ['endian-little', 'bit-32', 'ix86-common', 'common-linux', 'common-glibc', 'i586-linux', 'common']
DEBUG: Executing shell function do_compile
NOTE: make -j 9
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/uic ../phonesim-1.20/src/controlbase.ui -o src/ui_controlbase.h
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/control.h -o src/moc_control.cpp
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/phonesim.h -o src/moc_phonesim.cpp
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/hardwaremanipulator.h -o src/moc_hardwaremanipulator.cpp
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/callmanager.h -o src/moc_callmanager.cpp
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/simfilesystem.h -o src/moc_simfilesystem.cpp
/OE/build/wpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/qt5/moc ../phonesim-1.20/src/simapplication.h -o src/moc_simapplication.cpp
Could not create output file
Makefile:977: recipe for target 'src/ui_controlbase.h' failed
make: *** [src/ui_controlbase.h] Error 1
make: *** Waiting for unfinished jobs....
WARNING: /OE/build/wpb/webos-ports/tmp-glibc/work/i586-webos-linux/phonesim/1.20-r0/temp/run.do_compile.3359:1 exit 1 from 'exit 1'
ERROR: oe_runmake failed
ERROR: Function failed: do_compile (log file is located at /OE/build/wpb/webos-ports/tmp-glibc/work/i586-webos-linux/phonesim/1.20-r0/temp/log.do_compile.3359)

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 8790f63..ad4e512 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -67,6 +67,7 @@ QT_V_MOC_  = $(QT_V_MOC_$(AM_DEFAULT_VERBOSITY))
 QT_V_MOC_0 = @echo "  MOC     " $@;
 
 src/moc_%.cpp: src/%.h
+	$(MKDIR_P) src
 	$(QT_V_MOC)$(MOC) $< -o $@
 
 QT_V_UIC   = $(QT_V_UIC_$(V))
@@ -74,4 +75,5 @@ QT_V_UIC_  = $(QT_V_UIC_$(AM_DEFAULT_VERBOSITY))
 QT_V_UIC_0 = @echo "  UIC     " $@;
 
 src/ui_%.h: src/%.ui
+	$(MKDIR_P) src
 	$(QT_V_UIC)$(UIC) $< -o $@
